### PR TITLE
Add Ingress documentation

### PR DESCRIPTION
**Pull request by arturdzm**

### CODE REVIEW

File: doc/user-guide.adoc [hunk 1]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+kind: RuntimeComponent
+metadata:
+  name: my-app
+  namespace: backend
+spec:
+  applicationImage: quay.io/my-repo/my-app:1.0
+  expose: true
+  route:
+    annotations:
+      # this annotation can be used to specify name of ingress controller to use
```
Comment (chirp1) [inline doc/user-guide.adoc:581]: Change to "You can use annotations to specify the name of the ingress controller to use.".

File: doc/user-guide.adoc [hunk 2]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+      # there could be multiple installed for different types of incoming traffic (external DNS, internal DNS, etc)
```
Comment (chirp1) [inline doc/user-guide.adoc:582]: Change to "You can install multiple ingress controllers to address different types of incoming traffic such as an external or internal DNS.".

File: doc/user-guide.adoc [hunk 3]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+      kubernetes.io/ingress.class: "nginx" 
+      
+      # nginx ingress specific annotations, enbled secure pod connection
+      nginx.ingress.kubernetes.io/ssl-redirect: true
+      nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+
+      # traefik specific annotations, enbled secure pod connection
+      traefik.ingress.kubernetes.io/service.serversscheme: https
+
+    #use custom hostname for this Ingress
```
Comment (chirp1) [inline doc/user-guide.adoc:592]: Change to: "Use a custom hostname for the following ingress controller:".

File: doc/user-guide.adoc [hunk 4]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+    annotations:
+      # this annotation can be used to specify name of ingress controller to use
+      # there could be multiple installed for different types of incoming traffic (external DNS, internal DNS, etc)
+      kubernetes.io/ingress.class: "nginx" 
+      
+      # nginx ingress specific annotations, enbled secure pod connection
+      nginx.ingress.kubernetes.io/ssl-redirect: true
+      nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+
+      # traefik specific annotations, enbled secure pod connection
```
Comment (chirp1) [inline doc/user-guide.adoc:589]: Change to "The following traefik annotation enables a secure pod connection:".

File: doc/user-guide.adoc [hunk 5]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+spec:
+  applicationImage: quay.io/my-repo/my-app:1.0
+  expose: true
+  route:
+    annotations:
+      # this annotation can be used to specify name of ingress controller to use
+      # there could be multiple installed for different types of incoming traffic (external DNS, internal DNS, etc)
+      kubernetes.io/ingress.class: "nginx" 
+      
+      # nginx ingress specific annotations, enbled secure pod connection
```
Comment (chirp1) [inline doc/user-guide.adoc:585]: Change to "The following nginx annotation enables a secure pod connection:".

File: doc/user-guide.adoc [hunk 6]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+      nginx.ingress.kubernetes.io/ssl-redirect: true
+      nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+
+      # traefik specific annotations, enbled secure pod connection
+      traefik.ingress.kubernetes.io/service.serversscheme: https
+
+    #use custom hostname for this Ingress
+    host: app-v1.mycompany.com
+    #reference pre-existing TLS sercret
```
Comment (chirp1) [inline doc/user-guide.adoc:594]: Change to "Reference a pre-existing TLS secret:".

File: doc/user-guide.adoc [hunk 7]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
 
 _This feature is only available if you are running on OKD or OpenShift._
 
+
+==== Non-Knative deployment (Ingress)
+
+Before you can use the Ingress resource to expose your cluster, you must install an ingress controller, such a Nginx or Traeik..
```
Comment (leochr) [inline doc/user-guide.adoc:520]: - Traeik --> Traefik
- Extra `.` at the end.

File: doc/user-guide.adoc [hunk 8]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+
+_The Ingress resource is created only if the `Route` resource is not available._
+
+
+To use the Ingress resource, set the `defaultHostName` variable in the runtime-component-operator ConfigMap object to a host name such as mycomp.com
```
Comment (leochr) [inline doc/user-guide.adoc:525]: It'll be good to italicize values such as _runtime-component-operator_ and _mycomp.com_

File: doc/user-guide.adoc [hunk 9]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+
+
+===== Simple Ingress that uses `defaultHostName` and no `TLS`:
```
Comment (leochr) [inline doc/user-guide.adoc:528]: Suggestion: and no `TLS` --> without `TLS`

File: doc/user-guide.adoc [hunk 10]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+kind: RuntimeComponent
+metadata:
+  name: my-app
+  namespace: backend
+spec:
+  applicationImage: quay.io/my-repo/my-app:1.0
+  expose: true
+----
+
+With default hostname of mycompany.com, the application is available at the http://my-app-backend.mycompany.com URL.
```
Comment (leochr) [inline doc/user-guide.adoc:542]: Italicizing `mycompany.com` will be good.

File: doc/user-guide.adoc [hunk 11]
```diff
@@ -514,6 +514,87 @@ To create a secured HTTPS route, see the link:++#certificate-manager-integration
...
+spec:
+  applicationImage: quay.io/my-repo/my-app:1.0
+  expose: true
+  route:
+    certificate: {}
+----
+
+===== Advanced Ingress configuration:
+
+Most of the Ingress configuraiton is achieved through annotations. Annotations such as Nginx, HAProxy, Traefik, and others are specific to the ingress controller implementation.
```
Comment (leochr) [inline doc/user-guide.adoc:565]: configuraiton --> configuration

**Pull request merged**